addons/stock_transport/models/stock_picking_batch.py

```diff
@@ -0,0 +1,112 @@
+        self._update_sequences()
+        return res
+
+    def write(self, vals):
+        res = super().write(vals)
+        self._update_sequences()
+        return res
+
+    def _update_sequences(self):
+        sorted_records = self.picking_ids.sorted(key=lambda r: r.zip_code or '0')
+        for idx, record in enumerate(sorted_records):
+            record.sequence = idx
+
+    @api.depends('vehicle_id')
+    def _compute_vehicle_category_id(self):
+        for rec in self:
+            rec.vehicle_category_id = rec.vehicle_id.category_id if rec.vehicle_id else False
+
+    def _inverse_vehicle_category_id(self):
+        pass
+
+    @api.depends('vehicle_id', 'max_picking_weight', 'max_picking_volume')
+    def _compute_vehicle_details(self):
+        for record in self:
+            vehicle_details = []
+            if record.vehicle_id:
+                vehicle_details.append(record.vehicle_id.name)
+            vehicle_details.append(f"{round(record.max_picking_weight, 2)} Kg") if record.max_picking_weight else vehicle_details.append("0Kg")
+            vehicle_details.append(f"{round(record.max_picking_volume, 2)} m³") if record.max_picking_volume else vehicle_details.append("0m³")
+            record.vehicle_details = ', '.join(vehicle_details)
+
+    @api.depends('picking_ids.max_weight', 'vehicle_category_id.max_weight')
+    def _compute_used_weight_percentage(self):
+        for batch in self:
+            batch.max_picking_weight = sum(picking.max_weight for picking in batch.picking_ids)
+            batch_vehicle_max_weight = batch.vehicle_category_id.max_weight
+            used_weight_percentage = 100 * (batch.max_picking_weight / batch_vehicle_max_weight) if batch_vehicle_max_weight else 0.0
+            batch.used_weight_percentage = used_weight_percentage
+
+    @api.depends('picking_ids.max_volume', 'vehicle_category_id.max_volume')
+    def _compute_used_volume_percentage(self):
+        for batch in self:
+            batch.max_picking_volume = sum(picking.max_volume for picking in batch.picking_ids)
+            batch_vehicle_max_volume = batch.vehicle_category_id.max_volume
+            used_volume_percentage = 100 * (batch.max_picking_volume / batch_vehicle_max_volume) if batch_vehicle_max_volume else 0.0
+            batch.used_volume_percentage = used_volume_percentage
+
+    @api.depends('picking_type_id', 'max_picking_weight', 'max_picking_volume')
+    def _compute_picking_info(self):
+        for record in self:
+            picking_type = f"{record.picking_type_id.name}: " if record.picking_type_id else ''
+            max_picking_weight = f"{round(record.max_picking_weight, 2)}kg" if record.max_picking_weight else '0Kg'
+            max_picking_volume = f"{round(record.max_picking_volume, 2)}m³" if record.max_picking_volume else '0m³'
+            picking_info_str = f"{picking_type}{max_picking_weight}, {max_picking_volume}".rstrip(', ')
+            record.picking_info = picking_info_str
```

Comment:
You probably don't need a compute for that. Also kg and m³ should come from the system parameter. Some countries don't use metric system.

Reply:
we need this compute field for the Gantt view to display like `Delivery Order: 36.19kg, 3.6m3` as per the task spec.
Additionally, KG and m³ should comes from the system parameter that is right but for that I need to handle it with this approach. Please correct me if I am wrong 

i.e
Category   max_wight  max_weight_uom  volume  volume_uom
Mini Truck	750.00   	KG		         80.00      m³